src/Toolkit/Toolkit/UI/Controls/BookmarksView/BookmarksViewDataSource.cs

```diff
@@ -338,7 +338,11 @@ private void OnCollectionChanged(NotifyCollectionChangedEventArgs args)
         {
             RunOnUIThread(() =>
             {
+#if MAUI && __IOS__
+                CollectionChanged?.Invoke(this, new NotifyCollectionChangedEventArgs(NotifyCollectionChangedAction.Reset));
```

Comment:
I'm not quite following what this is doing? Why a full reset?

Reply:
There's a bug in the collectionview implementation on iOS, if the event is raised with an insert at the end of the list, e.g. at index 4, the collectionview implementation tries to access the collection at index 5, resulting in an exception. Doing a full reset each time the list changes fixes the problem.

Reply:
That happens if your notification change events doesn't match exactly what is actually in the collection. iOS is more picky about this, and it becomes trickier to handle with RunOnUIThread because to collection might change more.

Reply:
Can we at least add a //todo or an issue to address this correctly later?